Makefile.Protobuf.mk

```diff
@@ -0,0 +1,160 @@
+
+# Find all OTEL .proto files, remove leading path (only keep relevant namespace dirs).
+OTEL_PROTO_FILES=$(subst $(OTEL_PROTO_SRC_DIR)/,,\
+   $(shell ls $(OTEL_PROTO_SRC_DIR)/{common,resource,trace}/v1/*.proto))
+
+# Macro to execute a command passed as argument.
+# DO NOT DELETE EMPTY LINE at the end of the macro, it's required to separate commands.
+define exec-command
+$(1)
+
+endef
+
+# DO NOT DELETE EMPTY LINE at the end of the macro, it's required to separate commands.
+define print_caption
+  @echo "🏗️ "
+  @echo "🏗️ " $1
+  @echo "🏗️ "
+
+endef
+
+# Macro to compile Protobuf $(2) into directory $(1). $(3) can provide additional flags.
+# DO NOT DELETE EMPTY LINE at the end of the macro, it's required to separate commands.
+# Arguments:
+#  $(1) - output directory
+#  $(2) - path to the .proto file
+#  $(3) - additional flags to pass to protoc, e.g. extra -Ixxx
+#  $(4) - additional options to pass to gogo plugin
+define proto_compile
+  $(call print_caption, "Processing $(2) --> $(1)")
+
+  $(PROTOC) \
+    $(PROTO_INCLUDES) \
+    --gogo_out=plugins=grpc,$(strip $(4)),$(PROTO_GOGO_MAPPINGS):$(PWD)/$(strip $(1)) \
+    $(3) $(2)
+
+endef
+
+# TODO add proto-hotrod to the list after regenerating its file (may need linter tweaking)
+.PHONY: proto
+proto: proto-model proto-api-v2 proto-storage-v1 proto-zipkin proto-openmetrics proto-otel proto-api-v3
+
+.PHONY: proto-model
+proto-model:
+	$(call proto_compile, model, idl/proto/api_v2/model.proto)
+	$(PROTOC) -Imodel/proto --go_out=$(PWD)/model/ model/proto/model_test.proto
+
+.PHONY: proto-api-v2
+proto-api-v2:
+	$(call proto_compile, proto-gen/api_v2, idl/proto/api_v2/query.proto)
+	$(call proto_compile, proto-gen/api_v2, idl/proto/api_v2/collector.proto)
+	$(call proto_compile, proto-gen/api_v2, idl/proto/api_v2/sampling.proto)
+
+.PHONY: proto-openmetrics
+proto-openmetrics:
+	$(call print_caption, Processing OpenMetrics Protos)
+	$(foreach file,$(OPENMETRICS_PROTO_FILES),$(call proto_compile, proto-gen/api_v2/metrics, $(file)))
+	@# TODO why is this file included in model/proto/metrics/ in the first place?
+	rm proto-gen/api_v2/metrics/otelmetric.pb.go
+
+.PHONY: proto-storage-v1
+proto-storage-v1:
+	$(call proto_compile, proto-gen/storage_v1, plugin/storage/grpc/proto/storage.proto, -Iplugin/storage/grpc/proto)
+	$(PROTOC) \
+		-Iplugin/storage/grpc/proto \
+		--go_out=$(PWD)/plugin/storage/grpc/proto/ \
+		plugin/storage/grpc/proto/storage_test.proto
+
+.PHONY: proto-hotrod
+proto-hotrod:
+	$(call proto_compile, , examples/hotrod/services/driver/driver.proto)
+
+.PHONY: proto-zipkin
+proto-zipkin:
+	$(call proto_compile, proto-gen/zipkin, idl/proto/zipkin.proto, -Iidl/proto)
+
+# Target 'proto-prepare-otel' modifies OTEL proto to use proto-import path jaeger.proto.*
+# The modification is needed because OTEL collector already uses opentelemetry.proto.*
+# and two complied protobuf types cannot have the same import path. The root cause is that the compiled OTLP
```

Comment:
```suggestion
# and two compiled protobuf types cannot have the same import path. The root cause is that the compiled OTLP
```

Reply:
will fix in the next PR

Reply:
https://github.com/jaegertracing/jaeger/pull/5039